src/analyzer.ts

```diff
@@ -200,6 +203,29 @@ export class Analyzer {
   async analyze(
     root: ScriptReference,
     extraArgs: string[] | undefined
+  ): Promise<AnalyzeResult> {
+    this._logger?.log({
+      type: 'info',
+      detail: 'analysis-started',
+      script: root,
+    });
+    const analyzeResult = await this._actuallyAnalyze(root, extraArgs);
+    let rootScriptConfig = undefined;
+    if (analyzeResult.config.ok) {
+      rootScriptConfig = analyzeResult.config.value;
+    }
```

Comment:
This is very nice. What do you think about inlining this as a ternary in value position of `rootScriptConfig`? Then you can skip this if there isn't a logger.

Although it may be a bit less readable. Up to you.

Reply:
Ah, I like htat!